autoemulate/experimental/tuner.py

```diff
@@ -64,7 +64,12 @@ def run(self, model_class: type[Emulator]) -> tuple[list[float], list[ModelConfi
             }
 
             # TODO: consider whether to pass as tensors or dataloader
-            m = model_class(train_x, train_y, **model_config)
+            # TODO: is there a better way to distinguish between models that
+            # require training data for initialisation as well as fitting?
+            if "x" in model_class.__init__.__code__.co_varnames:
```

Comment:
Is it possible to do a check based on the different emulator types we've defined (e.g., is it a GP model or not)?

Reply:
Initially thumbed up, but actually I feel like your way would need to be updated when new emulators are added, but my way is more flexible, what do you think?

Reply:
We might prefer to not handle this at the caller level (i.e. instantiating the class) but instead within each class's `__init__`? With the aim that the init interface can be the same for all emulators.

For cases like LightGBM where the training data is not required for the init, `x` and `y` can be unused or e.g. `_, _ = x, y`.

In #384 ([code](https://github.com/alan-turing-institute/autoemulate/blob/5aeee8d7e23457a7b61a4dd0a678dc527808067d/autoemulate/experimental/emulators/base.py#L23-L25)), the base class's `__init__` added x, y for this purpose. This approach might need to change if we update the interfaces as part of #400 (e.g. with dataloaders instead of x, y).